docker/tests/factories/perplexity_response_factory.py

```
@@ -0,0 +1,91 @@
+    MockPerplexityClient,
+    MockPerplexityResponse,
+)
+
+
+class PerplexityResponseFactory:
+    """Factory for creating pre-configured Perplexity response mocks."""
+
+    @staticmethod
+    def successful_research(
+        content: str = "Python is a high-level programming language.",
+        citations: List[str] = None,
```

Ruff (0.13.3)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)